apps/web/src/pages/templates/workflow/TimedDigestMetadata.tsx

```diff
@@ -0,0 +1,155 @@
+import { Controller, useFormContext } from 'react-hook-form';
+import { Group } from '@mantine/core';
+import { format } from 'date-fns';
+import { DigestTypeEnum, DigestUnitEnum } from '@novu/shared';
+
+import { colors, Input, SegmentedControl } from '../../../design-system';
+import { inputStyles } from '../../../design-system/config/inputs.styles';
+import { WeekDaySelect } from './digest/WeekDaySelect';
+import { ScheduleMonthlyFields } from './digest/ScheduleMonthlyFields';
+import { When } from '../../../components/utils/When';
+
+const convertUnitToLabel = (unit: DigestUnitEnum) => {
+  switch (unit) {
+    case DigestUnitEnum.SECONDS:
+      return 'second';
+    case DigestUnitEnum.MINUTES:
+      return 'minute';
+    case DigestUnitEnum.HOURS:
+      return 'hour';
+    case DigestUnitEnum.DAYS:
+      return 'day';
+    case DigestUnitEnum.WEEKS:
+      return 'week';
+    case DigestUnitEnum.MONTHS:
+      return 'month';
+  }
+};
+
+export const TimedDigestMetadata = ({ index, readonly }: { index: number; readonly: boolean }) => {
+  const { control, watch, setValue } = useFormContext();
+  const unit: DigestUnitEnum =
+    watch(`steps.${index}.digestMetadata.${DigestTypeEnum.TIMED}.unit`) ?? DigestUnitEnum.DAYS;
+  const amountFieldName = `steps.${index}.digestMetadata.${DigestTypeEnum.TIMED}.${unit}.amount`;
+  const atTimeFieldName = `steps.${index}.digestMetadata.${DigestTypeEnum.TIMED}.${unit}.atTime`;
+  const amountDefaultValue = unit === DigestUnitEnum.MINUTES ? '5' : '1';
+  const atTimeDefaultValue = format(new Date(), 'HH:mm');
+
+  return (
+    <>
+      <Controller
+        control={control}
+        defaultValue={DigestUnitEnum.DAYS}
+        name={`steps.${index}.digestMetadata.${DigestTypeEnum.TIMED}.unit`}
+        render={({ field }) => {
+          return (
+            <SegmentedControl
+              {...field}
+              size="sm"
+              sx={{
+                maxWidth: '100% !important',
+              }}
+              fullWidth
+              disabled={readonly}
+              data={[
+                { value: DigestUnitEnum.MINUTES, label: 'Min' },
+                { value: DigestUnitEnum.HOURS, label: 'Hour' },
+                { value: DigestUnitEnum.DAYS, label: 'Daily' },
+                { value: DigestUnitEnum.WEEKS, label: 'Weekly' },
+                { value: DigestUnitEnum.MONTHS, label: 'Monthly' },
+              ]}
+              onChange={field.onChange}
+              data-test-id="digest-unit"
+            />
+          );
+        }}
+      />
+      <Group data-test-id="timed-group" spacing={8} sx={{ color: colors.B60 }}>
+        <span>Every</span>
+        <Controller
+          control={control}
+          key={amountFieldName}
```

Comment:
`key` is a trick to have a one "amount (every)" field and re-mount it when `amountFieldName` changes, and it changes when switching between the tabs: min, hours, days, weeks, months